### PR TITLE
Annotation fixes

### PR DESCRIPTION
**Pull request by jleft**
+ Improved Annotation visual test to test data changes and transitions
+ Decorate now has the appropriate selection passed to it
+ Added a key function property to allow the user to define how the data should be joined - is the proposed change a sensible default? Is there a better option?
+ I haven't made the changes to support fade in and out on the transition yet - I think a decision on #259 should be made first

### CODE REVIEW

File: visual-tests/test-scripts/tools/annotation.js [hunk 1]
```diff
@@ -96,9 +96,24 @@
...
+    setInterval(function() {
+        // Update an annotation
+        chartLayout.getPlotArea().select('#annotation')
+            .datum([randomValue()])
+            .call(annotation);
+
+        // Update an annotation with a transition
+        var decimalData = [randomValue(), randomValue(), randomValue()];
+        decimalData.splice(0, Math.floor(Math.random() * 2));
+        chartLayout.getPlotArea().select('#annotationDecimal')
```
Comment (chrisprice) [inline visual-tests/test-scripts/tools/annotation.js]: new lines on .

File: components/tools/annotation.js [hunk 2]
```diff
@@ -11,7 +11,7 @@
                 var container = d3.select(this);
 
                 // Create a group for each annotation
-                var g = fc.utilities.simpleDataJoin(container, 'annotation', data, annotation.yValue.value);
+                var g = fc.utilities.simpleDataJoin(container, 'annotation', data, annotation.keyFunction.value);
```
Comment (chrisprice) [inline components/tools/annotation.js]: keyValue

File: components/tools/annotation.js [hunk 3]
```diff
@@ -32,13 +32,14 @@
...
 
-                annotation.decorate.value(container);
+                annotation.decorate.value(g);
             });
         };
 
         annotation.xScale = fc.utilities.property(d3.time.scale());
         annotation.yScale = fc.utilities.property(d3.scale.linear());
         annotation.yValue = fc.utilities.functorProperty(fc.utilities.fn.identity);
+        annotation.keyFunction = fc.utilities.functorProperty(function(d, i) { return i; });
```
Comment (chrisprice) [inline components/tools/annotation.js]: move into fn

**Comment by jleft** [comment]
Thanks for the comments; amends have be made in ae49c09.

**Comment by chrisprice** [comment]
Thanks!

**Pull request merged**